File: leetcode-algorithms/020. Valid Parentheses/ValidParentheses.cs
public class Solution {
    public bool IsValid(string s) {
         if (String.IsNullOrEmpty(s)) {
                return true;
            }
            string Queue = "";
            char[] strs = s.ToCharArray();
            int i = 0;
            for (; i < strs.Length; i++) {
                switch (strs[i]) {
                    case '(':
                        Queue = String.Concat(strs[i],Queue);
                        break;
                    case '{':
                        Queue = String.Concat(strs[i], Queue);
                        break;
                    case '[':
                        Queue = String.Concat(strs[i], Queue);
                        break;
                    case ')':
                        if (!Queue.StartsWith("("))
                        {
                            return false;
                        }
                        else
                        {
                            Queue = Queue.Remove(0, 1);
                        }
                        break;
                    case '}':
                        if (!Queue.StartsWith("{"))
                        {
                            return false;
                        }
                        else
                        {
                            Queue = Queue.Remove(0, 1);
                        }
                        break;
                    case ']':
                        if (!Queue.StartsWith("["))
                        {
                            return false;
                        }
                        else
                        {
                            Queue = Queue.Remove(0, 1);
                        }
                        break;
                }
            }
            if (Queue.Length > 0)
                return false;
            else
                return true;
    }
}

File: leetcode-algorithms/020. Valid Parentheses/ValidParentheses.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if s == "":
            return True

        list1 = []
        for char in s:
            if char == "]":
                if len(list1) == 0 or list1.pop() != "[":
                    return False
            elif char == "}":
                if len(list1) == 0 or list1.pop() != "{":
                    return False
            elif char == ")":
                if len(list1) == 0 or list1.pop() != "(":
                    return False
            else:
                list1.append(char)
        if len(list1) == 0:
            return True
        else:
            return False